livekit-api/src/signal_client/signal_stream.rs

```diff
@@ -54,13 +54,20 @@ impl SignalStream {
     /// SignalStream will never try to reconnect if the connection has been
     /// closed.
     pub async fn connect(
-        url: url::Url,
+        mut url: url::Url,
     ) -> SignalResult<(
         Self,
         mpsc::UnboundedReceiver<Box<proto::signal_response::Message>>,
     )> {
         log::info!("connecting to SignalClient: {}", url);
 
+        // Automatically switch to websocket scheme when using http
+        if url.scheme() == "https" {
```

Comment:
is it better to copy the URL instead of mutating the input?

Reply:
the url is moved, so we can copy it without impacting the caller

Reply:
edit* not copy